src/components/FirebaseAuth.jsx

```diff
@@ -0,0 +1,59 @@
+import React, { Component } from 'react';
+
+import firebase from 'firebase';
+import StyleFirebaseAuth from 'react-firebaseui/StyledFirebaseAuth';
+
+firebase.initializeApp({
+  apiKey: 'AIzaSyAXDqsWK4quNVaf9-YV2e28NsxkfA9rzJA',
+  authDomain: 'dreamerscholars.firebaseapp.com',
+});
+
+class FirebaseAuth extends Component {
+  uiConfig = {
+    signInFlow: 'popup',
+    signInOptions: [
+      firebase.auth.GoogleAuthProvider.PROVIDER_ID,
+      firebase.auth.EmailAuthProvider.PROVIDER_ID,
+      firebase.auth.FacebookAuthProvider.PROVIDER_ID,
+    ],
+    credentialHelper: 'none', // hacky way to disable redirect on email login
+    callbacks: {
+      signInSuccessWithAuthResult: () => false,
+    },
+  }
+
+  constructor(props) {
+    // Required step: always call the parent class' constructor
+    super(props);
+
+    // Set the state directly. Use props if necessary.
+    this.state = {
+      isSignedIn: false,
+    };
+  }
+
+  componentDidMount = () => {
+    firebase.auth().onAuthStateChanged((user) => {
+      this.setState({ isSignedIn: !!user });
+    });
+  }
+
+  render() {
+    return (
+      <div>
+        {this.state.isSignedIn ? (
+          <span>
+            <button type="button" className="button" onClick={() => firebase.auth().signOut()}>Sign out!</button>
+          </span>
+        ) : (
+          <StyleFirebaseAuth
+            uiConfig={this.uiConfig}
+            firebaseAuth={firebase.auth()}
+          />
```

Comment:
I think it should be ok to write this in one line: 
```suggestion
          <StyleFirebaseAuth uiConfig={this.uiConfig} firebaseAuth={firebase.auth()} />
```

It doesn't over the line length limit.

Reply:
ok got it your right its at 90 length for the line. but I did notice line 46 is at 114 so ill make it into two lines